src/foam/nanos/notification/email/EmailTemplateUtil.js

```diff
@@ -0,0 +1,396 @@
+/**
+ * @license
+ * Copyright 2020 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.nanos.notification.email',
+  name: 'EmailTemplateUtil',
+
+  javaImports: [
+    'foam.dao.DAO',
+    'foam.lib.json.*',
+    'foam.lib.parse.*',
+    'foam.lib.parse.Action',
+    'foam.lib.parse.Grammar',
+    'java.lang.StringBuilder',
+    'java.util.HashMap',
+    'java.util.Map',
+    'static foam.mlang.MLang.EQ'
+
+  ],
+
+  properties: [
+    {
+      class: 'Object',
+      javaType: 'java.lang.StringBuilder',
+      name: 'sb',
+      javaFactory: `return new StringBuilder();`,
+      documentation: `this StringBuilder is used to store the final parsed value`
+    },
+    {
+      class: 'Object',
+      javaType: 'java.lang.StringBuilder',
+      name: 'finalSb',
+      javaFactory: `return new StringBuilder();`,
+      documentation: `this temporary StringBuilder with non-parsed values is used to join
+        two templates in case if one extends another`
+    },
+    {
+      class: 'Map',
+      name: 'values',
+      documentation: `map of values of values for simple values where {{ key }} will be replaced with value`
+    },
+    {
+      class: 'FObjectProperty',
+      of: 'foam.nanos.notification.email.EmailTemplate',
+      name: 'extendedEmailTemplate',
+      documentation: `If provided email template extends another email template,
+          it will be stored in this prop`
+    },
+    {
+      class: 'Object',
+      javaType: 'java.lang.StringBuilder',
+      name: 'blockContentSb',
+      javaFactory: `return new StringBuilder();`
+    },
+    {
+      class: 'FObjectProperty',
+      of: 'foam.lib.parse.Grammar',
+      name: 'grammar',
+      documentation: `grammar for parsing "{{ val }}", "{% if/else}}" symbols`,
+      javaFactory: `
```

Comment:
The problem with this approach is that it is fairly expensive to build a parser, and then the parser doesn't get fast until it is used, but because you never reuse it, it never gets fast. Could this be reused? There are a couple approaches: 1. move it to its own class and make it a singleton (many other parsers do this) 2. move this whole class to be a service and then only have one instance. Then you would have to remove the stringbuilder as a property.